### PR TITLE
docs: proctored exam testing instructions

### PR DESCRIPTION
**Pull request by ilee2u**
**JIRA:** Link to JIRA ticket

**Description:** Describe in a couple of sentences what this PR adds

**Author concerns:** List any concerns about this PR - inelegant 
solutions, hacks, quick-and-dirty implementations, concerns about 
migrations, etc.

**Dependencies:** dependencies on other outstanding PRs, issues, etc. 

**Installation instructions:** List any non-trivial installation 
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [inline docs/test_plan/proctored_exam.rst:69]: these tests could be a good place to cover other effects of a reset too. Why not test the learner's completion and problem is reset as well here? (same comment goes for the below tests)

**Review by Zacharis278 — COMMENTED** [review]
few comments to get started.

High level, it would be nice to make this less Proctorio specific. If we do have a few Proctorio specific cases we want to cover we keep that as a separate test. The user flow should _mostly_ be the same regardless of the tool used.

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [inline docs/test_plan/proctored_exam.rst:26]: I think these another set of cases here for submitting an attempt that is auto verified and one needs a second review.

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [inline docs/test_plan/proctored_exam.rst:125]: NIT: this one part of the test requires additional access. Going back the the section as a learner and ensuring you get the correct interstitial can test the reset happened without needing admin. (the tester already has to login as a learner to check completion in the next step.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:36]: No this is me getting mixed up with the exams dashboard for the proctoring software being a broken link at the moment. This shouldn't be broken ideally.

File: docs/test_plan/proctored_exam.rst
Comment (alangsto) [inline docs/test_plan/proctored_exam.rst:55]: Are there any tests for what to expect when an error occurs? For the proctoring flow, these all look like "happy path" tests.

File: docs/test_plan/proctored_exam.rst
Comment (alangsto) [inline docs/test_plan/proctored_exam.rst:27]: NIT: using a single dash makes the text appear smaller than the following test cases. If this is meant to be the overarching category, I feel like this text should be bigger.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:55]: Yes those should be here: https://github.com/edx/edx-exams/pull/205/files/d1455d82318fcaebb675be2eff36909ceee74c6e#diff-b5c1a36768f79123cc3158c9c56b1b72e50008252b380bec3d570b0a2ca0cc37R119. I'm realizing I need to revise these to say "As a student, error the exam, and as an admin, reset the exam attempt, then try the exam again as a student" so that we reflect what would happen IRL.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:27]: OOPS

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [reply docs/test_plan/proctored_exam.rst:26]: I still think we're missing these test cases ensuring a review by the proctoring provider is received and the attempt is updated. We kind of do that as a side effect of setup for testing the instructor dashboard but I think it's best to be explicit. If someone is just testing exam flow they should be validating this case.

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [reply docs/test_plan/proctored_exam.rst]: This is mostly just a NIT but bumping this comment because I think it's worth calling out specifically.

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [reply docs/test_plan/proctored_exam.rst:55]: I see we're testing the exams dashboard for error attempts but to Alie's point shouldn't we have a test of an error from the "testing the exam flow" perspective? Something to test that the learner gets an error interstitial / the learner cannot access the exam content after an error.

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:16]: I think that the order of these prerequisite steps is a little confusing. For example, you're enrolling a non-staff learner account in "this course" before you have set up a course with proctoring. Based on your instructions, you may set up an entirely new course in which the learner is not enrolled.

I'd recommend breaking this up into "Proctoring Set Up Steps" and "User Set Up Steps" to make this clearer.

Comment (MichaelRoytman) [reply docs/test_plan/proctored_exam.rst:16]: Do you have a particular environment in mind here (e.g. stage)? Stage will have proctoring set up in a way that someone just starting in their devstack will not, so I think we should either not make references to a particular environment and link out to documentation for each environment (e.g. stage/production and development), or specify that this is just for stage/production.

Comment (MichaelRoytman) [reply docs/test_plan/proctored_exam.rst:16]: "certificates are enabled" is a little vague and could mean a variety of things. I think we should be more specific here. Do you mean "where there is a verified track" or something else?

Comment (MichaelRoytman) [reply docs/test_plan/proctored_exam.rst:16]: I know we don't have official documentation for this yet, but it would be good to clarify what "an LTI proctoring provider chosen" means - how does one do this or verify this? Depending on which environment this documentation applies to, you could just refer to the Proctoring Settings Page and/or reference our [Local Development and LTI Configuration](https://2u-internal.atlassian.net/wiki/spaces/PT/pages/256737327/Local+Development+and+LTI+Configuration) document for the development environment.

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:23]: I would have no idea what this means if I wasn't already familiar with the event bus. How does one do this? For what it's worth, you don't make a reference to the logs again, so is this step really necessary?

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:7]: Neither of these links is used in this document.

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:49]: What setup? The steps in "Can't Start Exam Without Proctoring Software Running"?

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:76]: Where is this tab? How do I get to this tab? How do I know if I'm at the old or new exams dashboard?

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:87]: Out of curiosity, how does this work? Does Proctorio return a response this quickly? I know Proctortrack would take some time to return back a review.

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:89]: Would it be faster and less vendor specific to go into the admin dashboard and set the attempt's status to `second_review_required`? This also takes a way the need to be "suspicious enough" and avoids relying on the vendor's review process, which we don't need to be testing here. This applies to all the test cases that rely on a review coming in.

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:93]: Both in the dashboard and where else?

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:107]: What is this testing? Why don't we do this same step for "Can Reject/Verify "Review Required" attempts"?

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [inline docs/test_plan/proctored_exam.rst:114]:
```suggestion
#. A modal pops up upon clicking "Manual Review" that only should have the "Cancel" and "Verify" Buttons
```

**Review by MichaelRoytman — COMMENTED** [review]
I know I've left a lot of feedback. I read these instructions from the perspective of a team member with limited familiarity with LTI for proctoring, Proctorio, and how to set up proctoring. I think there there is a good amount of assumed knowledge in this document that not everyone will have, and my comments reflect that. I'm not marking my feedback as blocking. I know it's a pain, but I do think this document could really benefit from being a lot more explicit.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:16]: - Changed the order of the steps as you recommended (still need to push the change)
- I have local in mind, but based on [this thread](https://twou.slack.com/archives/C049EF6VCTB/p1699386718501819) it seems like it's also fair to mention that we do have a test course in stage (at least as of now) 
- I'm honestly not too sure how certificates work so if "verified track" makes more sense I can just put that.
- Assuming `edx-exams` ends up in Open edX, would it make sense to just move the instructions for [Local Development and LTI Configuration](https://2u-internal.atlassian.net/wiki/spaces/PT/pages/256737327/Local+Development+and+LTI+Configuration) to the README and just make that the source of truth?

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:23]: Ah yeah that's just something I forgot to delete. I agree with you on this so I'll remove it.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:7]: Ah these are copied over from [timed_exam.rst](https://github.com/edx/edx-exams/blob/main/docs/test_plan/timed_exam.rst), where it seems like they aren't used either. I see what you're saying though, I can have these links referenced in the setup instructions where it makes sense.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:87]: Usually takes no more than a minute from what I've seen on Stage.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:93]: Ah that's a typo

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:107]: I'm realizing this test is redundant because the status change in the UI only occurs if the status change in the backend was successful (this is done via "dispatch" in the exam dashboard's thunks), so I'm just going to remove it.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:88]: Added instructions on how to get to this tab + how to tell if it's the newer one.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst]: Yes :)

**Comment by ilee2u** [comment]
Made some changes based off @MichaelRoytman and @Zacharis278's comments, still have some work to do though.

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst]: Added instructions to have this as an option to just retaking the exam

File: docs/test_plan/proctored_exam.rst
Comment (ilee2u) [reply docs/test_plan/proctored_exam.rst:36]: I added a test to the Proctorio section about checking that a review of the exam exists. Regardless of whether or not we're using Proctorio, wouldn't the test that we received a review by a proctoring provider be checking the exams dashboard itself?

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [reply docs/test_plan/proctored_exam.rst]: for the local instructions you are correct and those do need to move as part of the release to openedx. We've got that noted as a requirement here: https://2u-internal.atlassian.net/wiki/spaces/PT/pages/626753683/Beta+Release+and+Beyond.

I do think these tests need to be done in a deployed environment and not Tutor/Devstack in order to say we're confident in releasing these features.

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [reply docs/test_plan/proctored_exam.rst:7]: I think this might be how they are referenced the other docs has them right next to the step to create a proctored exam. Maybe we just want to point back to this if these instructions are needed anywhere later in the doc. If you use headings you can add internal links within the doc like this https://github.com/openedx/edx-proctoring/blob/master/docs/testing/test_plan.md?plain=1#L220. That can also be good to avoid repeating yourself  if one test depends on the setup of another

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [reply docs/test_plan/proctored_exam.rst]: @MichaelRoytman I do think there's also going to be a gap until we get instructions in readthedocs. I don't think we should duplicate how to setup a proctored exam, a user, a proctoring provider, etc here and should just reference readthedocs where needed.

File: docs/test_plan/proctored_exam.rst
Comment (Zacharis278) [reply docs/test_plan/proctored_exam.rst:36]: I think the test I'm looking for is something that validates the ACS works as intended. So a test along the lines of "if a user submits an attempt without violation the proctoring software can update the attempt to verified" / "if a user submits with violations the proctoring software can update the attempt to review required".

I think that's a distinctly different test from testing that a verified/rejected/second review attempt can be reset in the UI since that testing instructor experience but the ACS is part of the learner / exam flow.

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [reply docs/test_plan/proctored_exam.rst:7]: Ah, I see. So it's like reference materials? My bad - got confused!

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [reply docs/test_plan/proctored_exam.rst]: Yeah, totally agree with linking out. I didn't mean to suggest we should duplicate - just that I felt like some information was missing, which can definitely be added by referencing other documentation, like "Set up a proctored exam: <link>."

File: docs/test_plan/proctored_exam.rst
Comment (MichaelRoytman) [reply docs/test_plan/proctored_exam.rst]: Oh, nice!

**Review by MichaelRoytman — COMMENTED** [review]
Thanks for addressing my first set of feedback.

I added some non-blocking feedback, which you can take or leave.

I agree with Zach regarding the ACS tests.

File: README.rst
Comment (ilee2u) [reply README.rst:62]: I'd say so. Since this is meant to be seen by Open Edx's community, I believe it makes sense for this public README to be the source of truth instead of our internal doc. I'll make sure to do that link once this is merged.

**Review by Zacharis278 — APPROVED** [review]
I'd say we're looking good. One quick thing I noticed, there's a parent readme for this folder that has a link to the other sets of tests. Can you add a link there? https://github.com/edx/edx-exams/tree/main/docs/test_plan

**Pull request merged**
